### ISSUE TITLE
torch.split on empty dim

### ISSUE THREAD

**Issue by ailzhang** [pytorch/xla#892]
```
import torch
import torch_xla

device = 'xla:0'
# device='cpu'

shape = (0, 1, 3, 0)
x = torch.rand(shape, device=device)
a = torch.split(x, 1, dim=0)
print(a)
```
XLA returns empty list. Should match cpu result.
